lib/views.rb

```diff
@@ -26,4 +29,53 @@ def menu
 		puts '                        C.Scores                      '
 		puts '__________________________________________________________'
         print "\n"
-end
\ No newline at end of file
+end
+
+def new_game(g) 
+  while true do
+    g.b.print_board(g.b.dim)
+    t = g.turn == "x" ? "1" : "2"
+    puts "Player#{t} choose a number in 0 - #{g.b.dim*g.b.dim - 1} or Q to quit!"   
+    cell = gets.chomp
+    if cell.upcase == "Q"
+       puts "Quit"
+       break
+    elsif g.b.change_mat(cell.to_i,g.turn)
+        if g.b.check_winner(g.turn)
+          g.b.print_board(g.b.dim)
+          puts "Player#{t} you won !"
+          g.score[t.to_i  - 1] += 1
+          break
+        elsif g.b.draw? 
+          puts "its a draw"
+          break
+        end 
+        g.turn = g.switch_turn(g.turn)      
+    end     
+  end   
+end  
```

Comment:
This block does a lot of things. Split in multiple methods. For example, one method that prompts the user for a number between 0 and 9 and returns the user input. 

Some of these methods should reside that will take care of the game flow.